data-tidy.qmd

```diff
@@ -569,8 +569,10 @@ Since you don't know how to work with this sort of data yet, you'll want to foll
 
 ```{r}
 df |> 
-  group_by(id, name) |> 
-  summarize(n = n(), .groups = "drop") |> 
+  summarize(
+    .by = c(id, name),
+    n = n(),
+  ) |> 
```

Comment:
This is one where I'd just write

```r
summarise(n = n(), .by = c(id, name))
```

I'm not sure we need to commit to up-front-`.by` religiously.

I feel it is most useful if you have multiple expressions (or one multiline expression) in your summarise/mutate call, and you want to pull it up front to ensure it doesn't get "lost" in the details

Otherwise my brain seems to want to type out the expression first, and the `.by` second, and I'm not sure how much I want to fight that

Reply:
I guess my comment above is related to this. `.by` upfront feels weird.